core/server/apps/amp/index.js

```diff
@@ -1,15 +1,26 @@
-var router           = require('./lib/router'),
+var router              = require('./lib/router'),
     registerAmpHelpers  = require('./lib/helpers'),
+    config              = require('../../config'),
+    ampIsEnabled;
 
-    // Dirty requires
-    config     = require('../../config');
+ampIsEnabled = function ampIsEnabled() {
```

Comment:
The clearer way in my opinion is to use middlewares, because separating logic bits is always more readable and makes the code cleaner. 

The most important thing in this PR is to fix the not working use case for subdirectories. The rest is optional, because of time priority 👍